chainstate/tx-verifier/src/transaction_verifier/timelock_check.rs

```diff
@@ -133,5 +153,60 @@ where
         }
     }
 
+    let outputs = match tx.outputs() {
+        Some(outputs) => outputs,
+        None => return Ok(()),
+    };
+
+    if let Some(output_check_required) = output_check_required {
+        let block_height = tx_source.expected_block_height();
+        for output in outputs {
+            match output {
+                TxOutput::Transfer(_, _)
+                | TxOutput::Burn(_)
+                | TxOutput::CreateStakePool(_, _)
+                | TxOutput::ProduceBlockFromStake(_, _)
+                | TxOutput::CreateDelegationId(_, _)
+                | TxOutput::DelegateStaking(_, _) => Ok(()),
+                TxOutput::LockThenTransfer(_, _, timelock) => match output_check_required {
+                    OutputTimelockCheckRequired::DelegationSpendMaturity => {
+                        let required =
+                            chain_config.as_ref().spend_share_maturity_distance(block_height);
+                        check_output_timelock(timelock, required)
+                    }
+                    OutputTimelockCheckRequired::DecommissioningMaturity => {
+                        let required =
+                            chain_config.as_ref().decommission_pool_maturity_distance(block_height);
+                        check_output_timelock(timelock, required)
+                    }
+                },
+            }?;
+        }
+    }
```

Comment:
Unrelated blocks of code should either be prefixed with comments or moved into their own functions. We should avoid throwing separable code consecutively after one another... it just leads to confusion. Maybe create a function that does this check, which can be tested individual and has its own explanation of what's going on.